gitops/utils/async_runner.py

```diff
@@ -14,43 +16,47 @@ def init_curses():
     curses.cbreak()
     # Turn off cursor
     curses.curs_set(0)
-    return stdscr
+    return (win, height, width)
+
+
+def addstr(win_info, x, y, text, color=0):
+    win, height, width = win_info
+    win.addstr(x, y, text, color)
+    win.clrtoeol()
+    win.clearok(1)
+    win.refresh(0, 0, 0, 0, height - 1, width - 1)
 
 
 async def run_tasks_async_with_progress(tasks):
-    stdscr = init_curses()
-    stdscr.addstr(0, 0, 'Your command is now running on the following servers:')
+    sem = asyncio.Semaphore(10)
+    win_info = init_curses(len(tasks) + 1)
+    addstr(win_info, 0, 0, f'Your command is now running on the following {len(tasks)} servers (may extend off bottom of terminal):')
```

Comment:
Gross lol. Idm going with it for now, but what if it only listed the 10 servers it was currently processing, and then somehow summarised everyone else that is pending/complete on one or two lines?